Source/WebCore/css/CSSFontFace.cpp

```diff
@@ -710,7 +710,7 @@ static Font::Visibility visibility(CSSFontFace::Status status, CSSFontFace::Font
     }
 }
 
-RefPtr<Font> CSSFontFace::font(const FontDescription& fontDescription, bool syntheticBold, bool syntheticItalic, ExternalResourceDownloadPolicy policy, const FontPaletteValues& fontPaletteValues)
+RefPtr<Font> CSSFontFace::font(const FontDescription& fontDescription, bool syntheticBold, bool syntheticItalic, ExternalResourceDownloadPolicy policy, const FontPaletteValues& fontPaletteValues, const Ref<FontFeatureValues> fontFeatureValues)
```

Comment:
Does it make sense to put palette-values and feature-values together in a struct and thread that struct around, rather than having many arguments passed around?

Reply:
I think that’s a good suggestion.

Separately, the argument type `const Ref<FontFeatureValues>` does not make sense. You could take const Ref<FontFeatureValues>&, or `Ref<FontFeatureValues>`, or `Ref<FontFeatureValues>&&`, or `FontFeatureValues&`. But the `const` doesn’t do anything useful. It’s like having an argument of type `const bool`.

Reply:
I guess it would. There is already the `FontCreationContext` class which kinda serves the same point but not in this part of the codebase.